### PR TITLE
Move conditioning

### PR DESCRIPTION
**Pull request by TrentHawkins**
Starting to work on the general condition of movement checks:
* Board edge detection,
* Enemy piece square occupancy (target),
* Friendly piece square occupancy (block).

This is a rather complex condition already, I see if it semantically splitable between the classes. This means that objects will end up playing ball with this condition, but that's the best I got so far.

### CODE REVIEW

**Comment by TrentHawkins** [comment]
Not sure you guys saw this. This is the latest work, so I will try to commit the corrections you proposed here.

**Comment by TrentHawkins** [comment]
I think I took care of most of the issue in #15. The branch was rebranded, because from this point on I started working on a new feature, before #15 ever got the change to be merged.

**Comment by TrentHawkins** [comment]
The next step is capturing and blocking logic, probably provideth by `Board`.

File: src/pieces.py [hunk 1]
```diff
@@ -204,14 +203,14 @@ class Range(Piece):
...
+    def legal_moves(self, square: Square, condition: Callable[[Square], bool]) -> set[Square]:
         super().legal_moves.__doc__
         squares = set()
-        for step in self.steps:  # For each direction.
-            advanced_square = square  # Start looking at advanced squares
-            while condition(square, step):  # While we don't hit something.
-                advanced_square += step  # Advance to the next square in line.
+        for move in self.steps:  # For each direction.
+            advanced_square = square + move  # Start looking at advanced squares
+            while condition(advanced_square):  # While we don't hit something.
```
Comment (prototact) [inline src/pieces.py:211]: I see you fixed it.

**Pull request closed without merging**